cirq/contrib/acquaintance/permutation.py

```diff
@@ -125,6 +130,16 @@ def _decompose_(self, qubits: Sequence[ops.QubitId]) -> ops.OP_TREE:
                     yield swap_gate(*qubits[i:i+2])
                     mapping[i], mapping[i+1] = mapping[i+1], mapping[i]
 
+    def __repr__(self):
+        return ('cirq.contrib.acquaintance.LinearPermutationGate('
+                '{!r}, {!r})'.format(self._permutation, self.swap_gate))
+
+    def __eq__(self, other):
```

Comment:
You also need `__ne__` and `__hash__`. Alternatively, you can use the `cirq.value_equality` decorator and implement `_value_equality_values_(self)`.

Reply:
Done.